pages/profile/index.js

```diff
@@ -42,6 +43,19 @@ module.exports = () => {
     next();
   });
 
+  app.use('/:profileId/permission', permissions('profile.permissions'),
+    (req, res, next) => {
+      Promise.resolve().then(() => {
+        if (req.user.profile.id === req.profileId) {
+          const err = new Error('Unauthorised');
+          err.status = 401;
```

Comment:
_Technically_ this should be a 403 rather than a 401, but it doesn't _really_ matter.

Reply:
Wait a min , why ? 401 Unauthorized vs 403 Forbidden ?

Reply:
401 is for when a person can't do a thing because we don't know enough about them or there is an action they need to perform first, i.e. they're not logged in.

403 means "we know exactly who you are, and you are not allowed to do this".

In this case we know and recognise the user, it's just that they're not allowed to perform the action in question, and there isn't anything they can do to be allowed to.

Reply:
Ah, I see, thank you.